File: rc-application/src/main/java/com/transkript/reportcard/business/service/i/StudentService.java
package com.transkript.reportcard.business.service.i;

import com.transkript.reportcard.api.dto.StudentDto;
import com.transkript.reportcard.api.dto.response.EntityResponse;
import com.transkript.reportcard.data.entity.Student;
import java.util.List;

public interface StudentService {
    
    EntityResponse addStudent( StudentDto studentDto);

    
    List<StudentDto> getAllDto();

    
    StudentDto getDto( Long id);

    void delete( Long id);

    
    Student getEntity( Long id);

    
    EntityResponse update( StudentDto studentDto);
}

File: rc-application/src/main/java/com/transkript/reportcard/business/service/i/StudentService.kt
package com.transkript.reportcard.business.service.i

import com.transkript.reportcard.api.dto.StudentDto
import com.transkript.reportcard.api.dto.response.EntityResponse
import com.transkript.reportcard.data.entity.Student

interface StudentService {
    fun addStudent(studentDto: StudentDto?): EntityResponse?
    val students: List<StudentDto?>?
    fun getStudent(id: Long?): StudentDto?
    fun deleteStudent(id: Long?)
    fun getEntity(id: Long?): Student?
    fun updateStudent(id: Long?, studentDto: StudentDto?): EntityResponse?
}
